src/data/restore.ts

```diff
@@ -179,7 +180,11 @@ const restoreElement = (
         verticalAlign: element.verticalAlign || DEFAULT_VERTICAL_ALIGN,
         containerId: element.containerId ?? null,
         originalText: element.originalText || element.text,
-        lineHeight: element.lineHeight || getLegacyLineHeightForText(element),
+        // if no lineHeight set, detect it from text height, provided `height`
+        // is non-0, otherwise use the default
+        lineHeight:
+          element.lineHeight ||
+          (element.height ? detectLineHeight(element) : DEFAULT_LINE_HEIGHT),
```

Comment:
when height is not defined or 0, we can't detect lineHeight from the element, so we want to use the default. This we can use when creating text elements programmatically.